pkg/ruler/remote/querier_test.go

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package remote
+
+import (
+	"context"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/gogo/protobuf/proto"
+	"github.com/golang/snappy"
+	"github.com/prometheus/prometheus/prompb"
+	"github.com/stretchr/testify/require"
+	"github.com/weaveworks/common/httpgrpc"
+)
+
+type mockRoundTripper func(ctx context.Context, req *httpgrpc.HTTPRequest) (*httpgrpc.HTTPResponse, error)
+
+func (r mockRoundTripper) RoundTrip(ctx context.Context, req *httpgrpc.HTTPRequest) (*httpgrpc.HTTPResponse, error) {
+	return r(ctx, req)
+}
+
+func TestQuerier_ReadPrometheusHTTPrefix(t *testing.T) {
+	var inReq *httpgrpc.HTTPRequest
+	mockTr := func(ctx context.Context, req *httpgrpc.HTTPRequest) (*httpgrpc.HTTPResponse, error) {
+		inReq = req
+
+		b, err := proto.Marshal(&prompb.ReadResponse{
+			Results: []*prompb.QueryResult{
+				{},
+			},
+		})
+		require.NoError(t, err)
+
+		return &httpgrpc.HTTPResponse{
+			Code: http.StatusOK,
+			Body: snappy.Encode(nil, b),
+		}, nil
+	}
+	q := New(mockRoundTripper(mockTr), "/prometheus", log.NewNopLogger())
+
+	_, err := q.Read(context.Background(), &prompb.Query{})
+	require.NoError(t, err)
+
+	require.NotNil(t, inReq)
+	require.Equal(t, "/prometheus/api/v1/read", inReq.Url)
```

Comment:
We don't test any other properties of the request or the result. Same comment for `TestQuerier_QueryPrometheusHTTPrefix`.